src/core/qgsexpressioncontext.cpp

```diff
@@ -762,6 +789,16 @@ class GetProcessingParameterValue : public QgsScopedExpressionFunction
 
 ///@endcond
 
+
+QgsExpressionContextScope *QgsExpressionContextUtils::formScope( const QgsFeature &formFeature )
+{
+  QgsExpressionContextScope *scope = new QgsExpressionContextScope( QObject::tr( "Form" ) );
+  scope->setFeature( formFeature );
```

Comment:
I think this can be neater:

    scope->setVariable( QStringLiteral("current_feature"), formFeature, true );

and then modify GetCurrentFormFieldValue to use 

   const QgsFeature feat( context->variable( QStringLiteral("current_feature") ) );

This avoids ANY chance that the feature will be overridden somewhere later in the context, and also means that a handy @current_feature variable will also be created